commands/auth.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) OpenFaaS Author(s) 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	scope         string
+	authURL       string
+	clientID      string
+	audience      string
+	listenPort    int
+	launchBrowser bool
+)
+
+func init() {
+	authCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	authCmd.Flags().StringVar(&authURL, "auth-url", "", "OAuth2 auth-url")
+	authCmd.Flags().StringVar(&clientID, "client-id", "", "OAuth2 client_id")
+	authCmd.Flags().IntVar(&listenPort, "listen-port", 31111, "OAuth2 local port for receiving cookie")
```

Comment:
A static port seemed to be needed to make an exact match on Auth0's redirect_uri ACL.